drivers/ata8510/ata8510.c

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright (C) 2020 Antonio Galea <antonio.galea@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ata8510
+ * @{
+ *
+ * @file
+ * @brief       Implementation of public functions for ATA8510 radio module
+ *
+ * @author      Antonio Galea <antonio.galea@gmail.com>
+ * @author      Roberto Asquini <bobasquins@gmail.com>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include "xtimer.h"
+#include "thread.h"
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+
+#include "ata8510.h"
+#include "ata8510_params.h"
+#include "ata8510_internals.h"
+
+#define ENABLE_DEBUG (0)
+/*
+ * Warning: to correctly display the debug messages in callbacks,
+ * add CFLAGS+=-DTHREAD_STACKSIZE_IDLE=THREAD_STACKSIZE_DEFAULT to the build
+ * command.
+ */
+#include "debug.h"
+
+#define SPIDEV               (dev->params.spi)
+#define CSPIN                (dev->params.cs_pin)
+#define INTPIN               (dev->params.int_pin)
+#define RESETPIN             (dev->params.reset_pin)
+#define SLEEPPIN             (dev->params.sleep_pin)
+
+static char stack[THREAD_STACKSIZE_DEFAULT];
+static void *_recv_thread(void *arg);
+
+static void _irq_handler(void *arg)
+{
+    ata8510_t *dev = (ata8510_t *)arg;
+
+    switch (dev->internal_state) {
+        case ATA8510_MODE_IDLE:
+            DEBUG("[ata8510] _irq_handler: event received in mode IDLE\n");
+            break;
+        case ATA8510_MODE_TX:
+            DEBUG("[ata8510] _irq_handler: event received in mode TX\n");
+            mutex_unlock(&(dev->tx_lock));
+            break;
+        case ATA8510_MODE_RX:
+            DEBUG("[ata8510] _irq_handler: event received in mode RX\n");
+            mutex_unlock(&(dev->rx_lock));
+            break;
+        case ATA8510_MODE_RX_POLLING:
+            DEBUG(
+                "[ata8510] _irq_handler: event received in mode RX POLLING\n"
+                );
+            mutex_unlock(&(dev->rx_lock));
+            break;
+    }
+}
+
+static void _send_command(ata8510_t *dev, const void *out,
+                          void *in, size_t len)
+{
+    if (spi_acquire(SPIDEV, CSPIN, SPI_MODE_0, dev->params.spi_clk) < 0) {
+        DEBUG("[ata8510] ERROR: Cannot acquire SPI bus!\n");
+    }
+    gpio_clear(CSPIN);
+    xtimer_usleep(1);
+    spi_transfer_bytes(SPIDEV, SPI_CS_UNDEF, true, out, in, len);
+    xtimer_usleep(1);
+    gpio_set(CSPIN);
```

Comment:
```suggestion
    spi_transfer_bytes(SPIDEV, SPI_CS_UNDEF, false, out, in, len);
```

does this still work?